prometheus/converters.go

```diff
@@ -0,0 +1,176 @@
+package prometheus
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/influxdata/influxdb/influxql"
+	"github.com/influxdata/influxdb/models"
+)
+
+//go:generate protoc -I$GOPATH/src -I. --gogofaster_out=. remote.proto
```

Comment:
In influxql, we just use `--gogo_out`. Are those different generators and is this adding a new dependency? We likely should avoid using multiple protoc generators if we can. Also, is the `-I` needed? I don't see any included dependencies in the proto file.

Reply:
`gogofaster` is a superset of functionality of `gogo` with additional optimizations in the generated code and less pointer fields. This generally makes the structs easier and more efficient to work with in Go.

Reply:
The protobuf wire format is unchanged

Reply:
Should we be using it instead of `gogo` everywhere and avoid having multiple generators? I'm not opposed to switching generators, I would just like it if we used one. It also doesn't have to be this PR that switches everything to a new generator.

Reply:
Agree and we should move to the new generators. In many cases, we will need to do a significant amount of refactoring but the effort will be worth while for performance and ease of use.

Reply:
Just to make sure the other part of the comment isn't lost, is the `-I` needed?

Reply:
ok, so use the vanilla generator for now and we can evaluate gogofaster later?

Reply:
Go with `gogo_faster`. We are using it for the new storage APIs and should plan on migrating others over. It does not introduce additional dependencies. The `-I` includes mean that if future versions of the .proto file import other proto files, you won't have to update the `generate` line, so it is safe to leave it in.

Reply:
ok, should be all moved around and good now